### PR TITLE
Java zoned date time support

### PR DESCRIPTION
**Pull request by jannhendrik**

### CODE REVIEW

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/DateConversionConstants.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.data.cosmosdb.core.convert;
+
+public class DateConversionConstants {
+
+    public static final String ISO_8601_COMPATIBLE_DATE_PATTERN = "yyyy-MM-dd'T'HH:mm:s:SSSXXX";
```
Comment (Incarnation-p-lee) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/DateConversionConstants.java]: we have one class named `Constants` already, looks you can put the String there.

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/MappingDocumentDbConverter.java [hunk 2]
```diff
@@ -24,14 +25,21 @@
...
 import java.util.Date;
 
+import static com.microsoft.azure.spring.data.cosmosdb.core.convert.DateConversionConstants.ISO_8601_COMPATIBLE_DATE_PATTERN;
+
 public class MappingDocumentDbConverter
         implements EntityConverter<DocumentDbPersistentEntity<?>, DocumentDbPersistentProperty, Object, Document>,
         ApplicationContextAware {
 
+
+
```
Comment (Incarnation-p-lee) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/MappingDocumentDbConverter.java]: could you please help to remove the duplicated empty line.

File: src/test/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/ZonedDateTimeDeserializerTest.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+    @Test
+    public void parse() throws IOException {
+        final JsonParser jsonParser = Mockito.mock(JsonParser.class);
+        when(jsonParser.getValueAsString()).thenReturn(EXAMPLE_DATE_STRING);
+
+        final ZonedDateTime zonedDateTime = new ZonedDateTimeDeserializer().parse(jsonParser);
+
+        assertThat(zonedDateTime.equals(EXPECTED_SERIALIZED_ZONED_DATE_TIME)).isTrue();
+    }
```
Comment (Incarnation-p-lee) [inline src/test/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/ZonedDateTimeDeserializerTest.java]: For more professional advice, you can add one test for testing Exception when parsing.

File: src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/ZonedDateTimeDeserializer.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+public class ZonedDateTimeDeserializer extends JsonDeserializer<ZonedDateTime> {
+
+    @Override
+    public ZonedDateTime deserialize(final JsonParser jsonParser, DeserializationContext deserializationContext)
+            throws IOException {
+        return parse(jsonParser);
+    }
+
+    @VisibleForTesting
+    ZonedDateTime parse(final JsonParser jsonParser) throws IOException {
```
Comment (Incarnation-p-lee) [inline src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/convert/ZonedDateTimeDeserializer.java]: explicit modifier ?

**Comment by kxmas** [comment]
This change overrides the ability to customize the object mapper as described in this project's README.md

```Configurable ObjectMapper bean with unique name cosmosdbObjectMapper, only configure customized ObjectMapper if you really need to. e.g.,
   @Bean(name = "cosmosdbObjectMapper")
   public ObjectMapper objectMapper() {
      return new ObjectMapper(); // Do configuration to the ObjectMapper if required
   }
```

You should make the changes you want during initialization, not run time.

**Comment by kxmas** [comment]
Please revert this change.

com.microsoft.azure.spring.data.cosmosdb.core.convert.ObjectMapperFactory provides support for the Java8 date time classes

**Pull request merged**